pkg/processor/procbuiltin/unwrap_test.go

```diff
@@ -0,0 +1,246 @@
+		   },
+		   "before": null,
+		   "op": "c",
+		   "source": {
+		     "opencdc.readAt": "1674061777225877000",
+		     "opencdc.version": "v1"
+		   },
+		   "transaction": null,
+		   "ts_ms": 1674061777225
+		 },
+		 "schema": {} 
+		}`
+
+func TestUnwrap_Config(t *testing.T) {
+	tests := []struct {
+		name    string
+		config  processor.Config
+		wantErr bool
+	}{
+		{
+			name:    "empty config",
+			config:  processor.Config{},
+			wantErr: true,
+		},
+		{
+			name: "invalid config",
+			config: processor.Config{
+				Settings: map[string]string{"format": "bar"},
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid debezium config",
+			config: processor.Config{
+				Settings: map[string]string{"format": "debezium"},
+			},
+			wantErr: false,
+		},
+		{
+			name: "valid kafka-connect config",
+			config: processor.Config{
+				Settings: map[string]string{"format": "kafka-connect"},
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := UnwrapBuilder(tt.config)
+			if (err != nil) != tt.wantErr {
+				t.Fatalf("UnwrapBuilder() error = %v, wantErr = %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestUnwrap_Process(t *testing.T) {
+	tests := []struct {
+		name    string
+		record  record.Record
+		want    record.Record
+		config  processor.Config
+		wantErr bool
+	}{
+		{
+			name: "raw payload",
+			config: processor.Config{
+				Settings: map[string]string{"format": "debezium"},
+			},
+			record: record.Record{
+				Key: record.RawData{
+					Raw: []byte(`{"payload":"id"}`),
+				},
+				Position: []byte("position"),
+				Payload: record.Change{
+					Before: nil,
+					After: record.RawData{
+						Raw: []byte(DebeziumRecord),
+					},
+				},
+			},
+			want: record.Record{
+				Operation: record.OperationCreate,
+				Metadata: map[string]string{
+					"opencdc.readAt":  "1674061777225877000",
+					"opencdc.version": "v1",
+				},
+				Key: record.RawData{
+					Raw: []byte("id"),
+				},
+				Position: []byte("position"),
+				Payload: record.Change{
+					Before: record.StructuredData(nil),
+					After:  record.StructuredData{"description": "test1", "id": float64(27)},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "structured payload",
+			config: processor.Config{
+				Settings: map[string]string{"format": "debezium"},
+			},
+			record: record.Record{
+				Metadata: map[string]string{
+					"conduit.version": "v0.4.0",
+				},
+				Payload: record.Change{
+					Before: nil,
+					After: record.StructuredData{
+						"payload": map[string]interface{}{
+							"after": map[string]interface{}{
+								"description": "test1",
+								"id":          27,
+							},
+							"before": interface{}(nil),
+							"op":     "u",
+							"source": map[string]interface{}{
+								"opencdc.readAt":  "1674061777225877000",
+								"opencdc.version": "v1",
+							},
+							"transaction": interface{}(nil),
+							"ts_ms":       float64(1674061777225),
```

Comment:
If `opencdc.readAt` wouldn't be set in `source` then this value would be used to set `opencdc.readAt` in metadata, right? Can we change the test to test this please?

Reply:
`opencdc.readAt` should be a nano second value, right?

Reply:
Yeah, I expect the metadata field to contain `1674061777225000000`.